src/RadixDlt.NetworkGateway.PostgresIntegration/DbQueryExtensions.cs

```diff
@@ -79,7 +84,12 @@ public static IQueryable<LedgerTransaction> GetTopLedgerTransaction<TDbContext>(
             .Take(1);
     }
 
-    public static IQueryable<LedgerTransaction> GetLatestLedgerTransactionBeforeStateVersion<TDbContext>(this TDbContext dbContext, long beforeStateVersion)
+    /// <summary>
+    /// The long/horrible/verbose name is on purpose, to draw your attention and prevent people foot-gunning themselves.
+    /// A LedgerTransaction row contains large blobs, so you must SELECT the fields you need after using this, and not pull down the whole
+    /// ledger transaction row, to avoid possible performance issues.
+    /// </summary>
+    public static IQueryable<LedgerTransaction> GetLatestLedgerTransactionBeforeStateVersion__EnsureYouSelectRequiredFieldsAsRowsCanBeVeryBig<TDbContext>(this TDbContext dbContext, long beforeStateVersion)
```

Comment:
this extension is used only in one place, maybe better to just inline it in plece where it's used?